functions/_common/urlTools.ts

```diff
@@ -0,0 +1,12 @@
+export function getGrapherFilters(
+    searchParams: URLSearchParams
+): Record<string, unknown> {
+    const params = searchParams.size
+        ? Object.fromEntries(searchParams)
+        : undefined
+    // delete url query params that the download api uses but that are not related to grapher
+    delete params.v1
+    delete params.csvType
+    delete params.useColumnShortNames
```

Comment:
Would it be worth the extra effort to use a whitelist here (and in the creation of `urlWithFilters`) ? Otherwise you can add arbitrary text to the query string and get a readme that says whatever anyone wants.

![image](https://github.com/user-attachments/assets/3698b8a9-2d88-4baf-b244-342ba1786e23)

Reply:
Good point! It means that when we do this for MDims we'll have to expand the whitelist but it's probably the better way to do things.